dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/ingest/IngestUtilTest.java

```diff
@@ -6,320 +6,173 @@
 
-        // set metadata and add verson
-        FileMetadata fmd1 = new FileMetadata();
-        fmd1.setId(1L);
-        fmd1.setLabel("datafile1.txt");
+        List<DataFile> dataFileList = new ArrayList<>();
+        DataFile datafile1 = createDataFile(dataset, "application/octet-stream", "datafile1.txt");
+        FileMetadata fmd1 = createFileMetadata(datasetVersion, datafile1, 1L, "datafile1.txt");
         fmd1.setDirectoryLabel("");
-        fmd1.setDataFile(datafile1);
         datafile1.getFileMetadatas().add(fmd1);
         datasetVersion.getFileMetadatas().add(fmd1);
-        fmd1.setDatasetVersion(datasetVersion);
 
         dataFileList.add(datafile1);
 
-        DataFile datafile2 = new DataFile("application/octet-stream");
-        datafile2.setStorageIdentifier("datafile2.txt");
-        datafile2.setFilesize(200);
-        datafile2.setModificationTime(new Timestamp(new Date().getTime()));
-        datafile2.setCreateDate(new Timestamp(new Date().getTime()));
-        datafile2.setPermissionModificationTime(new Timestamp(new Date().getTime()));
-        datafile2.setOwner(dataset);
-        datafile2.setIngestDone();
-        datafile2.setChecksumType(DataFile.ChecksumType.SHA1);
-        datafile2.setChecksumValue("Unknown");
-
-        // set metadata and add version
-        FileMetadata fmd2 = new FileMetadata();
-        fmd2.setId(2L);
-        fmd2.setLabel("datafile2.txt");
+        DataFile datafile2 = createDataFile(dataset, "application/octet-stream", "datafile2.txt");
+        FileMetadata fmd2 = createFileMetadata(datasetVersion, datafile2, 2L, "datafile2.txt");
         fmd2.setDirectoryLabel("");
-        fmd2.setDataFile(datafile2);
         datafile2.getFileMetadatas().add(fmd2);
         datasetVersion.getFileMetadatas().add(fmd2);
-        fmd2.setDatasetVersion(datasetVersion);
 
         dataFileList.add(datafile2);
 
+        //WHEN
         IngestUtil.checkForDuplicateFileNamesFinal(datasetVersion, dataFileList);
 
-        boolean file1NameAltered = false;
-        boolean file2NameAltered = false;
-        for (DataFile df : dataFileList) {
-            if (df.getFileMetadata().getLabel().equals("datafile1-1.txt")) {
-                file1NameAltered = true;
-            }
-            if (df.getFileMetadata().getLabel().equals("datafile2-1.txt")) {
-                file2NameAltered = true;
-            }
-        }
-
-        // check filenames are unique and unaltered
-        assertEquals(file1NameAltered, true);
-        assertEquals(file2NameAltered, true);
+        //THEN
+        assertAll(
+                () -> assertTrue(dataFileList.stream().anyMatch(fileMetadataHasLabel("datafile1-1.txt"))),
+                () -> assertTrue(dataFileList.stream().anyMatch(fileMetadataHasLabel("datafile2-1.txt")))
+        );
 
         // try to add data files with "-1" duplicates and see if it gets incremented to "-2"
         IngestUtil.checkForDuplicateFileNamesFinal(datasetVersion, dataFileList);
```

Comment:
Similar to above, copy and then quadruple assertion.

Reply:
Done